src/price-spread-ratio/price-spread-ratio.js

```diff
@@ -0,0 +1,166 @@
+
+    // load the array of contract addresses, abis, and ethers interfaces
+    data.contractsData = common.getContractAddressesAbis(data.contractNames);
+
+    // create an ethers contract object for each contract and store with the other contract data
+    // also extract the FTX url corresponding to each address and store it with the contract data
+    data.contractsData.forEach((contractData) => {
+      // create and store the ethers contract
+      contractData.contract = new ethers.Contract(
+        contractData.address,
+        contractData.contractAbi,
+        data.provider,
+      );
+
+      // get the FTX url and store it
+      contractData.ftxUrl = data.config[contractData.name].ftxUrl;
+
+      // get the upper and lower limits of the price spread ratio and store them as BigNumbers
+      const contractName = contractData.name;
+      contractData.upperLimitPercent = new BigNumber(data.config[contractName].upperLimitPercent);
+      contractData.lowerLimitPercent = new BigNumber(data.config[contractName].lowerLimitPercent);
+
+      // store the time threshold for how long the price spread ratio must be outside the limits
+      contractData.timeThresholdSeconds = BigInt(data.config[contractName].timeThresholdSeconds);
+    });
+    /* eslint-enable no-param-reassign */
+  };
+}
+
+function provideHandleBlock(data) {
+  return async function handleBlock(blockEvent) {
+    const findings = [];
+
+    const blockTimestamp = BigInt(blockEvent.block.timestamp);
+
+    // iterate over the contracts of interest to get the current prices
+    const promises = data.contractsData.map(async (contractData) => {
+      // do not create a finding the first time the handler is run
+      // instead, initialize the stored timestamp for this contract
+      if (!contractData.lastTimestampInLimits) {
+        // eslint-disable-next-line no-param-reassign
+        contractData.lastTimestampInLimits = blockTimestamp;
+        return;
+      }
+
+      const { contract } = contractData;
+
+      // get the price for each asset from the FTX
+      const ftxResponse = await axios.get(contractData.ftxUrl);
+
+      // check that we received a status 200
+      if (ftxResponse.status !== 200) {
+        return;
+      }
+      // convert price to a BigNumber
+      const ftxPriceEth = new BigNumber(ftxResponse.data.result.price);
+      // convert price to Wei to compare with price returned by contract method
+      const ftxPriceWei = ftxPriceEth.times(WEI_PER_ETHER);
+
+      // get the price for each asset from the contract
+      const interval = 0;
+      // returned value is an ethers BigNumber
+      const perpPrice = await contract.getIndexPrice(interval);
+      // convert the price to a BigNumber
+      const perpPriceWei = new BigNumber(perpPrice.toString());
+
+      // calculate the price spread ratio
+      const priceSpreadRatio = (perpPriceWei.minus(ftxPriceWei)).dividedBy(ftxPriceWei);
```

Comment:
out of curiosity, is there a specific design reason to use the FTX price as the base rather than the perp fi price?